airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/StreamManager.kt

```diff
@@ -124,10 +126,11 @@ class DefaultStreamManager(
         return recordCount.get()
     }
 
-    override fun markEndOfStream(): Long {
+    override fun markEndOfStream(complete: Boolean): Long {
         if (markedEndOfStream.getAndSet(true)) {
```

Comment:
This should probably fail if we've received either state. Maybe make `receivedComplete` an `isComplete: CompletableDeferred<Boolean>`?

`completableDeferred.complete(...)` will return `false` if it's already been completed, so you could do `if(!isComplete.complete(complete))` lol

Reply:
Tho that would make `isComplete` suspend, which would be confusing. You could also do an `AtomicReference<Bool>(null)` to have a non-blocking tristate